app/cluster/dynamic.go

```diff
@@ -125,6 +163,15 @@ func (d *Dynamic) stop() {
 	d.serverStopCountStat.Increment()
 }
 
+func (d *Dynamic) handleWorkspaceChange(workspaces string) error {
+	if d.currentWorkspaceIDs == workspaces {
```

Comment:
should we have a validator for workspaces?

Reply:
Not sure what kind of validator we can do for workspaceIDs, we are pretty ok to accept any value as far as understand. Maybe repeated values could be a problem ...

cc @cisse21

Reply:
repeated values too should not be a problem but not sure what validations can be done if required as all of the data resides on control plane and we would have to request control plane to validate them(which is the source for scheduler anyways)

Reply:
So I am going without a validator here.

Reply:
One case which we might need to consider here is that the workspace string need not match but might have the same workspace ID's where this check would fail

Reply:
I was thinking to introduce [sets](https://github.com/deckarep/golang-set) for storing and comparing workspaceIDs.

Reply:
That would be better than a string comparison and could deal with both duplicates and IDs stored in a different order (e.g. `1,2` != `2,1`). Before going down that road I'd do a bunch of benchmarks to compare at least 2/3 different approaches for our use case.